contrib/qubes/src/fwupd_receive_updates.py

```diff
@@ -114,7 +114,7 @@ def _extract_archive(self, archive_path, output_path):
         archive_path -- absolute path to archive file
         output_path -- absolute path to the output directory
         """
-        cmd_extract = ["gcab", "-x", f"--directory={output_path}", f"{archive_path}"]
+        cmd_extract = ["cabextract", "-d", output_path, "-s", "--", archive_path]
```

Comment:
Good catch!  In that case I am almost certain I want gcab.  That said, if you are interested in rewriting some stuff in <insert memory safe language of choice>, I would be happy to help.

(One of the main considerations is that Qubes OS is stuck on an EOL Fedora in dom0 for most of its release cycle, so the only things in dom0 that are allowed to touch untrusted input are a small set of low-risk tools.  We even wrote our own RPM validator because we didn’t want to rely on RPM 4.14.x.)